go/enclave/rpc/TenStorageRead.go

```diff
@@ -0,0 +1,118 @@
+package rpc
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ten-protocol/go-ten/go/common/gethencoding"
+	"github.com/ten-protocol/go-ten/go/common/log"
+	"github.com/ten-protocol/go-ten/go/common/privacy"
+	"github.com/ten-protocol/go-ten/go/common/syserr"
+	gethrpc "github.com/ten-protocol/go-ten/lib/gethfork/rpc"
+)
+
+type StorageReadWithBlock struct {
+	address     common.Address
+	storageSlot string
+	block       *gethrpc.BlockNumberOrHash
+}
+
+func TenStorageReadValidate(reqParams []any, builder *CallBuilder[StorageReadWithBlock, string], _ *EncryptionManager) error {
+	// Parameters are [TransactionArgs, BlockNumber, 2 more which we don't support yet]
+	if len(reqParams) < 2 || len(reqParams) > 3 {
+		builder.Err = fmt.Errorf("unexpected number of parameters")
+		return nil
+	}
+
+	addressHex, ok := reqParams[0].(string)
+	if !ok {
+		builder.Err = fmt.Errorf("address not provided in parameters")
+		return nil
+	}
+
+	address := common.HexToAddress(addressHex)
+	slot, ok := reqParams[1].(string)
+	if !ok {
+		builder.Err = fmt.Errorf("storage slot not provided in parameters")
+		return nil
+	}
+
+	//todo: @siliev - this whitelist creation every time is bugging me
+	whitelist := privacy.NewWhitelist()
```

Comment:
it can be a service in the enclave, initialised at start time, and then available